app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,46 @@
+class BookingsController < ApplicationController
+  before_action :set_booking, only: [:show]
+
+  def new
+    @booking = Booking.new
+    @puppy = Puppy.find(params[:puppy_id])
```

Comment:
What about moving this line in a private method + before action [:new, :show, :confirm]?